### PR TITLE
accounts db/refactor accounts db test - convert test_full_clean_refcount

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem

- add full_clean_refcount tests for both account storage formats

Continued on #786

#### Summary of Changes

Fixes #

### CODE REVIEW

**Comment by HaoranYi** [comment]
The actual change is in the last commit.
Please only review the last change if you have already reviewed https://github.com/anza-xyz/agave/pull/786

**Review by brooksprumo — COMMENTED** [review]
Please re-request a review once the dependent PRs have been merged and this PR has been rebased.

**Comment by HaoranYi** [comment]
No code changes. Rebase on top of master to pick up https://github.com/anza-xyz/agave/pull/786.

File: accounts-db/src/accounts_db.rs [hunk 1]
```diff
@@ -12749,24 +12746,17 @@ pub mod tests {
         assert_eq!(accounts.ref_count_for_pubkey(&pubkey3), 0);
     }
 
-    #[test]
-    fn test_full_clean_refcount() {
```
Comment (jeffwashington) [inline accounts-db/src/accounts_db.rs:12753]: I think we keep this comment.

Comment (HaoranYi) [reply accounts-db/src/accounts_db.rs:12753]: yeah.

File: accounts-db/src/accounts_db.rs [hunk 2]
```diff
@@ -12749,24 +12746,25 @@ pub mod tests {
...
+    // do stores with a 4mb size so only 1 store is created per slot
+    define_accounts_db_test!(test_full_clean_refcount_no_first_4m, |accounts| {
+        do_full_clean_refcount(accounts, false, 4 * 1024 * 1024);
+    });
 
-        // do stores with a 4k size and store pubkey1 first
-        do_full_clean_refcount(false, 4096);
+    // do stores with a 4k size and store pubkey1 first
+    define_accounts_db_test!(test_full_clean_refcount_no_first_4k, |accounts| {
+        do_full_clean_refcount(accounts, true, 4 * 1024);
```
Comment (brooksprumo) [inline accounts-db/src/accounts_db.rs]: Should this one be `false` and the one below be `true`? That's what the original code does, and also seems to be how these are tests are named here, right? (`no_first` vs `first`?)

Comment (HaoranYi) [reply accounts-db/src/accounts_db.rs]: yeah. fixed.

**Review by brooksprumo — APPROVED** [review]
:shipit:

**Review by jeffwashington — APPROVED** [review]
lgtm

**Pull request merged**